tests/defer/src/commonTest/kotlin/test/DeferNormalizedCacheTest.kt

```diff
@@ -383,19 +387,16 @@ class DeferNormalizedCacheTest {
     // - an exception happens
     // - fallback to the cache
     // - because of the error the cache is missing some fields, so we get a cache miss
-    var throwable: Throwable? = null
-    val networkActual = apolloClient.query(WithFragmentSpreadsQuery()).toFlow()
-        .catch { t ->
-          throwable = t
-        }
+    val responses = apolloClient.query(WithFragmentSpreadsQuery()).toFlow()
         .toList()
 
-    assertResponseListEquals(networkResponses, networkActual)
-    assertIs<ApolloCompositeException>(throwable)
-    throwable as ApolloCompositeException
-    assertIs<ApolloNetworkException>(throwable!!.suppressedExceptions.first())
-    assertIs<CacheMissException>(throwable!!.suppressedExceptions.getOrNull(1))
-    assertEquals("Object 'computers.0.screen' has no field named 'isColor'", throwable!!.suppressedExceptions.getOrNull(1)!!.message)
+    assertResponseListEquals(networkResponses, responses.take(2))
+
+    assertIs<ApolloNetworkException>(responses.get(2).exception)
+    responses.get(3).exception.apply {
+      assertIs<CacheMissException>(this)
+      assertEquals("Object 'computers.0.screen' has no field named 'isColor'", message)
+    }
```

Comment:
That one is interesting. The previous behaviour is not emulated by our `toFlowV3` replacement in the `@defer` case. Here 3.x had a `ApolloCompositeException` even though there were technically success network responses.